static/app/views/performance/transactionSummary/transactionEvents/content.tsx

```diff
@@ -0,0 +1,246 @@
+
+    return (
+      <Fragment>
+        <TransactionHeader
+          eventView={transactionsListEventView}
+          location={location}
+          organization={organization}
+          projects={projects}
+          transactionName={transactionName}
+          currentTab={Tab.Events}
+          hasWebVitals={
+            getCurrentLandingDisplay(location, projects, eventView).field ===
+            LandingDisplayField.FRONTEND_PAGELOAD
+          }
+          handleIncompatibleQuery={handleIncompatibleQuery}
+        />
+        <Layout.Body>
+          <Layout.Main fullWidth>
+            <Search {...this.props} />
+            <StyledTable>
+              <DiscoverQuery
+                location={location}
+                eventView={transactionsListEventView}
+                orgSlug={organization.slug}
+                limit={limit}
+                cursor={cursor}
+                referrer="api.discover.transactions-list"
+              >
+                {({isLoading, pageLinks, tableData}) => {
+                  return (
+                    <React.Fragment>
+                      <TransactionsTable
+                        eventView={eventView}
+                        organization={organization}
+                        location={location}
+                        isLoading={isLoading}
+                        tableData={tableData}
+                        columnOrder={eventView.getColumns()}
+                        titles={transactionsListTitles}
+                        handleCellAction={this.handleCellAction}
+                        generateLink={{
+                          id: generateTransactionLink(transactionName),
+                          trace: generateTraceLink(
+                            eventView.normalizeDateSelection(location)
+                          ),
+                        }}
+                      />
+                      <Pagination
+                        pageLinks={pageLinks}
+                        onCursor={this.handleCursor}
+                        size="small"
+                      />
+                    </React.Fragment>
+                  );
+                }}
+              </DiscoverQuery>
+            </StyledTable>
+          </Layout.Main>
+        </Layout.Body>
+      </Fragment>
+    );
+  }
+}
+
+function generateTraceLink(dateSelection) {
+  return (
+    organization: Organization,
+    tableRow: TableDataRow,
+    _query: Query
+  ): LocationDescriptor => {
+    const traceId = `${tableRow.trace}`;
+    if (!traceId) {
+      return {};
+    }
+
+    return getTraceDetailsUrl(organization, traceId, dateSelection, {});
+  };
+}
+
+function generateTransactionLink(transactionName: string) {
+  return (
+    organization: Organization,
+    tableRow: TableDataRow,
+    query: Query
+  ): LocationDescriptor => {
+    const eventSlug = generateEventSlug(tableRow);
+    return getTransactionDetailsUrl(organization, eventSlug, transactionName, query);
+  };
+}
```

Comment:
Would these be better in the `transactionSummary/transactionEvents/utils.tsx` file?

Reply:
Agreed, move to `utils.tsx`!